PennMobile/Dining/Networking + Cache/LocalJSONStore.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  LocalJSONStore.swift
+//  PennMobile
+//
+//  Created by Dominic Holmes on 10/22/19.
+//  Copyright © 2019 PennLabs. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - LocalJSONStore
+// This class can be used by any Codable struct as a simple cacheing layer. Check out DiningDataStore.swift for an example.
+class LocalJSONStore<T> where T : Codable {
+    let storageType: DataStoreType
+    let filename: String
+    
+    init(storageType: DataStoreType, filename: String) {
+        self.storageType = storageType
+        self.filename = filename
+        ensureFolderExists()
+    }
+    
+    // MARK: - Saving
+    func save(_ object: T) {
+        do {
+            let data = try JSONEncoder().encode(object)
+            try data.write(to: fileURL)
+        } catch let e {
+            print("CACHE ERROR: \(e)")
```

Comment:
Let's remove this print statement. Do we want to handle failure?

Reply:
Added a comment here explaining when it fails (when we screw up our encoding funcs).

I'm thinking we should leave this in, as otherwise we might silently break our Cache in the future.

Reply:
I guess this could also fail when data.write fails, but we can't do anything about that. It's also just a cache, so okay if it fails to write once.. it'll just be overwritten next time.

Though if our encoding is messed up, it'll fail forever until we fix it.